tests/scripts/spack_manifest/test_getter.py

```diff
@@ -0,0 +1,480 @@
+        root_spec_getter = RootSpec.from_file(manifest_path)
+
+        assert root_spec_getter.root_spec == "access-om2@git.2024.03.0=latest", "Root spec should be correctly retrieved from file."
+
+    def test_from_file__invalid(self):
+        manifest_path = "tests/scripts/spack_manifest/injection/inputs/invalid_manifest.yaml"
+
+        with pytest.raises(OSError):
+            RootSpec.from_file(manifest_path)
+
+    def test_get_name__valid(self):
+        manifest = {
+            "spack": {
+                "specs": ["access-om2@git.1.0.0"]
+            }
+        }
+
+        root_spec_getter = RootSpec(manifest)
+        name = root_spec_getter.get_name()
+
+        assert name == "access-om2", "Root spec name should be correctly extracted."
+
+    def test_get_name__invalid(self):
+        manifest = {
+            "spack": {
+                "specs": ["@git.1.0.0"]
+            }
+        }
+
+        root_spec_getter = RootSpec(manifest)
+        with pytest.raises(NoSectionComponentError):
+            root_spec_getter.get_name()
+
+    def test_get_ref__valid_git(self):
+        manifest = {
+            "spack": {
+                "specs": ["access-om2@git.1.0.0"]
+            }
+        }
+
+        root_spec_getter = RootSpec(manifest)
+        ref = root_spec_getter.get_ref()
+
+        assert ref == "1.0.0", "Root spec ref should be correctly extracted."
+
+    def test_get_ref__valid_non_git(self):
+        manifest = {
+            "spack": {
+                "specs": ["access-om2@1.0.0"]
+            }
+        }
+
+        root_spec_getter = RootSpec(manifest)
+        ref = root_spec_getter.get_ref()
+
+        assert ref == "1.0.0", "Root spec ref should be correctly extracted."
+
+    def test_get_ref__valid_git_with_constraints(self):
+        manifest = {
+            "spack": {
+                "specs": ["access-om2@git.1.0.0 %compiler@2.0.0"]
+            }
+        }
+
+        root_spec_getter = RootSpec(manifest)
+        ref = root_spec_getter.get_ref()
+
+        assert ref == "1.0.0", "Root spec ref should be correctly extracted with constraints."
+
+    def test_get_ref__valid_git_with_space_constraints(self):
+        manifest = {
+            "spack": {
+                "specs": ["access-om2@git.1.0.0 %compiler@2.0.0 +debug"]
+            }
+        }
+
+        root_spec_getter = RootSpec(manifest)
+        ref = root_spec_getter.get_ref()
+
+        assert ref == "1.0.0", "Root spec ref should be correctly extracted with space constraints."
+
+    def test_get_ref__valid_git_with_spack_version_and_constraints(self):
+        manifest = {
+            "spack": {
+                "specs": ["access-om2@git.1.0.0=develop %compiler@2.0.0 +debug"]
+            }
+        }
+
+        root_spec_getter = RootSpec(manifest)
+        ref = root_spec_getter.get_ref()
+
+        assert ref == "1.0.0", "Root spec ref should be correctly extracted with Spack version and constraints."
```

Comment:
This could be collapsed into a single parameterized test but that may be at the loss of a fine-grained error message.

If you did go that way, would be nice to have a version of the tests that collapsed the whitespace and confirmed they worked as expected.